tools/tools.go

```diff
@@ -47,36 +47,19 @@ import (
 
 // Tools will represent the installed tools
 type Tools struct {
-	Directory   string
-	IndexURL    string
-	LastRefresh time.Time
-	Logger      func(msg string)
-	installed   map[string]string
-	mutex       sync.RWMutex
+	Directory string
+	Index     *index.Resource
+	Logger    func(msg string)
+	installed map[string]string
+	mutex     sync.RWMutex
 }
 
 // Init creates the Installed map and populates it from a file in .arduino-create
-func (t *Tools) Init(APIlevel string) {
-	createDir(t.Directory)
+func (t *Tools) Init() {
 	t.mutex.Lock()
 	t.installed = make(map[string]string)
 	t.mutex.Unlock()
 	t.readMap()
-	t.mutex.RLock()
-	if t.installed["apilevel"] != APIlevel {
-		t.mutex.RUnlock()
-		// wipe the folder and reinitialize the data
-		os.RemoveAll(t.Directory)
-		createDir(t.Directory)
-		t.mutex.Lock()
-		t.installed = make(map[string]string)
-		t.installed["apilevel"] = APIlevel
-		t.mutex.Unlock()
-		t.writeMap()
-		t.readMap()
-	} else {
-		t.mutex.RUnlock()
-	}
 }
```

Comment:
```suggestion
type Tools struct {
	directory string
	index     *index.Resource
	logger    func(msg string)
	installed map[string]string
	mutex     sync.RWMutex
}

func New(directory string, index *index.Resource, logger func(msg string)) *Tools {
	t := &Tools{
		directory: directory,
		index:     index,
		logger:    logger,
		installed: map[string]string{},
		mutex:     sync.RWMutex{},
	}
	_ = t.readMap()
	return t
}
```
nit: would be great to make such properties unexported and use a "constructor" function
I see that we populate a global variable so it might require a bit more effort. Up to you if you want to refactor this